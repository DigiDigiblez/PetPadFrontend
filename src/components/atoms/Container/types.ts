import { ReactNode } from "react";

export interface IContainerProps {
    children: ReactNode;
    className: string;
}
