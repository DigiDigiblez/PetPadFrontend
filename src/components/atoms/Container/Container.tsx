import "./Container.scss";

import React from "react";

import { IContainerProps } from "./types";

const Container = ({ children, className }: IContainerProps) => {
    return <div className={className}>{children}</div>;
};

export default Container;
