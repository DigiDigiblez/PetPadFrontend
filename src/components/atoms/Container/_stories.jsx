import { storiesOf } from "@storybook/react";
import React from "react";

import { LayoutDecorator } from "../../../../.storybook/decorators";
import Text from "../Text";
import Container from "./Container";

storiesOf("Atoms/Table", module)
    .addDecorator(LayoutDecorator)
    .add("with Text", () => (
        <Container>
            <Text>This is a container</Text>
        </Container>
    ));
