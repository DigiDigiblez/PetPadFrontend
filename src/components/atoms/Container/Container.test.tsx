import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import Container from "./Container";

describe("Given the Table component", () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Container>Hello world!</Container>);
    });

    it("should render the expected markup", () => {
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
