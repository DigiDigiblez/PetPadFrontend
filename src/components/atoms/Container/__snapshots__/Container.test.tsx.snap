// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Table component should render the expected markup 1`] = `
<div
  className="container"
>
  Hello world!
</div>
`;
